test/unit/deadline_mcp/test_mcp.py

```
@@ -0,0 +1,113 @@
+    """Test utility functions."""
+
+    def test_default_serializer(self):
+        """Test serializer with different object types."""
+        assert _default_serializer("string") == "string"
+        assert _default_serializer(123) == "123"
+
+        class TestObj:
+            def __init__(self):
+                self.attr = "value"
+
+        assert _default_serializer(TestObj()) == {"attr": "value"}
+
+    def test_default_error_handler(self):
+        """Test error handler formats errors correctly."""
+        error = Exception("test error")
+        result = _default_error_handler(error)
+
+        assert result["error"] == "test error"
+        assert result["type"] == "Exception"
+
+    def test_create_wrapper(self):
+        """Test wrapper creation and execution."""
+
+        def mock_func(param1: str, param2: int = 10):
+            return {"param1": param1, "param2": param2}
+
+        config = {"func": mock_func, "params": ["param1", "param2"], "description": "Test function"}
+
+        wrapper = _create_wrapper(config, _default_serializer, _default_error_handler)
+
+        assert callable(wrapper)
+        assert hasattr(wrapper, "__signature__")
+
+        result = wrapper(param1="test", param2=20)
+        assert isinstance(result, dict)
+
+
+class TestRegisterAPITools:
+    """Test API tools registration."""
+
+    def test_basic_registration(self):
+        """Test registering valid tools."""
+        mock_app = MagicMock()
+
+        # Clean up any existing markers
+        if hasattr(api.list_farms, "_mcp_tool_registered"):
+            delattr(api.list_farms, "_mcp_tool_registered")
+
+        register_api_tools(mock_app, [api.list_farms])
+
+        assert mock_app.tool.call_count == 1
+        assert hasattr(api.list_farms, "_mcp_tool_registered")
+
+        # Clean up
+        delattr(api.list_farms, "_mcp_tool_registered")
+
+    def test_invalid_tools_skipped(self):
+        """Test that invalid tools are skipped."""
```

Should this raise an exception instead of skipping? By skipping, I think errors could slip into the implementation more easily.

---

Thanks, yes makes sense to raise exception instead of skipping.